### ISSUE TITLE
Problem syncing utf-8 chars

### ISSUE THREAD

**Issue by ricardoamaro** [org-trello/org-trello#379]
Hello,

While syncing any Trello card the second time from my board i get weird characters instead of punctuation.

Example before:
![image](https://user-images.githubusercontent.com/384962/37044381-3a70d1a4-215b-11e8-9250-c46c1ace5ff9.png)

Example After 2 syncs:

![image](https://user-images.githubusercontent.com/384962/37045322-93b0268c-215d-11e8-8c20-6e8669397142.png)

Result in emacs:

![image](https://user-images.githubusercontent.com/384962/37045354-a4c5ad98-215d-11e8-9064-fcf58268fc8f.png)

* DONE Teste configura��o UTF8
  :PROPERTIES:
  :orgtrello-id: 5a9ec14382e9095e57443192
  :orgtrello-local-checksum: 1cc9efe7739f711f774fcff0ede78877c9bd59a196ed8bddf55b078c2b414884
  :END:

## bug?

I believe it's a bug

### Expected behavior

I expected the characters to remain the same

### Actual behavior

There is a broken behaviour of character conversion:

* DONE Teste configuração UTF8

turns into:

* DONE Teste configura��o UTF8

### Steps to reproduce the behavior

2x
```
  (org-trello-check-setup)
  (org-trello-update-board-metadata)
  (org-trello-sync-buffer)
  (org-trello-sync-buffer 'from)
```

### More information on

http://org-trello.github.io/contributions.html#issues

Thank you
Ricardo

**Comment by ardumont** [comment]
Looks like the same thing as #377.

**Comment by ricardoamaro** [comment]
hello @ardumont isn't this the same issue mentioned in https://github.com/org-trello/org-trello/issues/318#issuecomment-190739525 ?

there was a function there: 
```
(defun orgtrello-query--decode-as-utf-8 ()
  "Permit to decode the buffer's response data as utf-8."
  (let ((data (buffer-string)))
    (erase-buffer)
    (insert (decode-coding-string data 'utf-8))
    (message "########## org-trello trace\n%s" (buffer-string))
    (goto-char (point-min))
    (orgtrello-log-msg orgtrello-log-debug "Decoded data to utf-8")))
```

**Comment by ardumont** [comment]
Yes, correct. It's unicode issue alright.

> there was a function there:

Right, I don't find it anymore, so possibly a regression happened.

Thanks for raising this up!

**Comment by ardumont** [comment]
@PierreTechoueyres kindly reminded me of his fix #373 integrated in my 0.8.1 branch (not yet merged).

```
(defun orgtrello-query--post-or-put (server query-map &optional success-callback error-callback authentication-p)
  "Execute the POST/PUT request to SERVER with QUERY-MAP with optional SUCCESS-CALLBACK, ERROR-CALLBACK and AUTHENTICATION-P."
  (with-local-quit
    (let ((uri           (->> query-map orgtrello-data-entity-uri (orgtrello-query--compute-url server)))
          (entity-method (orgtrello-data-entity-method query-map))
          (params        (when authentication-p (orgtrello-query--authentication-params)))
          (parser        'orgtrello-query--http-parse)
          (headers       '(("Content-type" . "application/json")))
          (data          (encode-coding-string (->> query-map orgtrello-data-entity-params json-encode) 'utf-8))
          (success-cbck  (if success-callback success-callback 'orgtrello-query--standard-success-callback))
          (error-cbck    (if error-callback error-callback 'orgtrello-query--standard-error-callback)))
      (if (orgtrello-data-entity-sync query-map)
          (request uri
                   :sync    t
                   :type    entity-method
                   :params  params
                   :headers headers
                   :data    data
                   :parser  parser
                   :success success-cbck
                   :error   error-cbck)
        `(deferred:$
           (request-deferred ,uri
                             :type    ,entity-method
                             :params  (quote ,params)
                             :headers (quote ,headers)
                             :data    ,data
                             :parser  (quote ,parser))
           (deferred:nextc it
             ,success-cbck)
           (deferred:error it
             ,error-cbck))))))
```

Can you please try and apply this to your local install and check whether you still see your error after that?

Install: <kbd>M-: <paste-snippet> RET</kbd>

Thanks

Cheers,

**Comment by ricardoamaro** [comment]
Did:
0. Installed that function on my init file to be evaluated after org-trello
```
  (with-eval-after-load 'org-trello
    (defun orgtrello-query--post-or-put (server query-map &optional success-callback error-callback authentication-p)
  "Execute the POST/PUT request to SERVER with QUERY-MAP with optional SUCCESS-CALLBACK, ERROR-CALLBACK and AUTHENTICATION-P."
  (with-local-quit
    (let ((uri           (->> query-map orgtrello-data-entity-uri (orgtrello-query--compute-url server)))
          (entity-method (orgtrello-data-entity-method query-map))
          (params        (when authentication-p (orgtrello-query--authentication-params)))
          (parser        'orgtrello-query--http-parse)
          (headers       '(("Content-type" . "application/json")))
          (data          (encode-coding-string (->> query-map orgtrello-data-entity-params json-encode) 'utf-8))
          (success-cbck  (if success-callback success-callback 'orgtrello-query--standard-success-callback))
          (error-cbck    (if error-callback error-callback 'orgtrello-query--standard-error-callback)))
      (if (orgtrello-data-entity-sync query-map)
          (request uri
                   :sync    t
                   :type    entity-method
                   :params  params
                   :headers headers
                   :data    data
                   :parser  parser
                   :success success-cbck
                   :error   error-cbck)
        `(deferred:$
           (request-deferred ,uri
                             :type    ,entity-method
                             :params  (quote ,params)
                             :headers (quote ,headers)
                             :data    ,data
                             :parser  (quote ,parser))
           (deferred:nextc it
             ,success-cbck)
           (deferred:error it
             ,error-cbck))))))
    )
```

1. I created a new card on trello with the correct chars.
2. Installed the board from scratch
3. ran
  (org-trello-sync-buffer)
  (org-trello-sync-buffer 'from)

Got 
* TODO Teste configuração UTF8

Running a couple more times the problem seems to be gone.
I think this fixes the issue.
Thank you @ardumont @PierreTechoueyres

**Comment by ardumont** [comment]
Awesome!

Thanks, that confirms @PierreTechoueyres's insight and patch ;)

Again awesome!

**Comment by ardumont** [comment]
Should be fixed in 0.8.1.
